homeassistant/components/nam/sensor.py

```diff
@@ -0,0 +1,93 @@
+"""Support for the Nettigo Air Monitor service."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import NAMDataUpdateCoordinator
+from .const import DOMAIN, SENSORS
+
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add a Nettigo Air Monitor entities from a config_entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    sensors = []
+    for sensor in SENSORS:
+        if sensor in coordinator.data:
+            sensors.append(NAMSensor(coordinator, sensor))
```

Comment:
No warning or anything (else part)?

Reply:
This is a DIY device so it can have different sensors. We check if the sensor is available in data from the device, if it is, we create an entity for it. Warning is not needed here in my opinion.

Reply:
good argument.